platform/ios/src/MGLMapView.mm

```diff
@@ -1299,8 +1304,122 @@ - (void)didMoveToSuperview
     [super didMoveToSuperview];
 }
 
+
+- (UIInterfaceOrientationMask)applicationSupportedInterfaceOrientations {
+    static UIInterfaceOrientationMask orientationMask = UIInterfaceOrientationMaskAll;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        
+        // "The system intersects the view controller's supported orientations with
+        // the app's supported orientations (as determined by the Info.plist file or
+        // the app delegate's application:supportedInterfaceOrientationsForWindow:
+        // method) and the device's supported orientations to determine whether to rotate.
+        
+        UIApplication *application = [UIApplication sharedApplication];
+        
+        if (self.window && [application.delegate respondsToSelector:@selector(application:supportedInterfaceOrientationsForWindow:)]) {
```

Comment:
So `applicationSupportedInterfaceOrientations` is perhaps better named `windowSupportedInterfaceOrientations`?

Since it’s possible (AirPlay, external monitors, CarPlay — though probably rare?) for a view to change windows, it seems like we might want to listen to [`-[UIView willMoveToWindow:]`](https://developer.apple.com/documentation/uikit/uiview/1622563-willmovetowindow?language=objc) and update the value of the `applicationSupportedInterfaceOrientations` property.

Reply:
Done.